pkg/component/extensions/extension/extension_test.go

```diff
@@ -322,7 +396,7 @@ var _ = Describe("Extension", func() {
 				time.Microsecond*400,
 				time.Second,
 			)
-			Expect(ext.DestroyBeforeKubeAPIServer(ctx)).To(MatchError(error(&multierror.Error{Errors: []error{fakeError, fakeError}})))
```

Comment:
I'm struggling to understand why there is another expected error in this test case, and if this test is correct or not.
When calling `DestroyBeforeKubeAPIServer` there should not be any new error with this PR, correct?

EDIT: ah, this is because there is no `DestroyAfterWorker`, and we simply delete the `AfterWork` extensions together with `AfterKubeAPIServer`. Correct?

Reply:
`AfterWorker` is only available during reconcile. In this test suite, the `afterWorkerExtension` does not specify a behavior for migrate and delete, so the defaults should be used (same as `defaultExtension`), which for delete this is `BeforeKubeAPIServer`.